File: Proxy/PHP/ProxyNetwork.php
<?php

require_once "INetwork.php";

class ProxyNetwork implements INetwork {
    private INetwork $network;
    private $bannedSites = [
        "abc.com",
        "xyz.com"
    ];

    public function __construct(INetwork $network)
    {
        $this->network = $network;
    }

    public function connect(string $sites) {
        if (in_array($sites, $this->bannedSites)) {
            echo "Sites " . $sites . " Blocked\n";
            return;
        }

        $this->network->connect($sites);
    }
}

File: Proxy/Dart/ProxyNetwork.dart
import 'INetwork.dart';

class ProxyNetwork implements INetwork {
  INetwork? _network;
  List<String> _bannedSites = ["abc.com", "xyz.com"];

  ProxyNetwork(this._network);

  @override
  void connect(String sites) {
    if (_bannedSites.contains(sites)) {
      print("Site $sites blocked");
      return;
    }

    _network!.connect(sites);
    
  }
}
